### PR TITLE
[SVCS-499] Adding `size_as_int` property to `BaseFileMetadata`

### PR DESCRIPTION
**Pull request by AddisonSchiller**
refs: https://openscience.atlassian.net/browse/SVCS-499

## Purpose
Some provider metadata returns size as a string, not an int. Add a property that returns size as an int.

## Summary of Changes
Added `size_as_int` field to `BaseFileMetadata`. 
Added explicit metadata tests for the new property

## Test/QA notes

This change is very minimal and should require little QA. It should be most notable in the tests for now.

### CODE REVIEW

File: waterbutler/core/metadata.py [hunk 1]
```diff
@@ -265,6 +265,14 @@ def size(self) -> int:
         """ Size of the file in bytes. """
         raise NotImplementedError
 
+    @property
+    def size_as_int(self) -> int:
+        """ Size of the file in bytes. """
```
Comment (Johnetordoff) [inline waterbutler/core/metadata.py]: There should be a comment here as well as in BaseMetadata.size that these to co-exsist to maintain backwards compatibility.

File: tests/providers/bitbucket/test_metadata.py [hunk 2]
```diff
@@ -35,6 +35,8 @@ def test_build_file_metadata(self, file_metadata, owner, repo):
         assert metadata.created_utc is None
         assert metadata.content_type is None
         assert metadata.size == 13
+        assert metadata.size_as_int == 13
+        assert type(metadata.size_as_int) == int
```
Comment (Johnetordoff) [inline tests/providers/bitbucket/test_metadata.py]: Generally you want to avoid comparisons of this nature in Python because of how it handles inheritance. Typically we'd use `assert isinstance(metadata.size_as_int, int)`.

**Comment by Johnetordoff** [comment]
LGTM! 👍

File: waterbutler/core/metadata.py [hunk 3]
```diff
@@ -265,6 +265,17 @@ def size(self) -> int:
...
         raise NotImplementedError
 
+    @property
+    def size_as_int(self) -> int:
+        """ Size of the file as an int.
+        Some providers give metadata as an int. Both exist to maintain backwards
+        compatibility.
+        """
+        if self.size is not None:
+            return int(self.size)
```
Comment (cslzchen) [inline waterbutler/core/metadata.py]: Is it possible that `self.size` is an empty String? 
```bash
>>> int('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: ''
```
I prefer an easier solution by adding a `try ... except ...` around the method.
A less recommended one (more complicated + performance overhead) is to check the type first and handle `string` and `int` separately.

Comment (cslzchen) [reply waterbutler/core/metadata.py]: And we should add tests for this case as well.

Comment (Johnetordoff) [reply waterbutler/core/metadata.py]: Good catch, this is kind of a weird conditional. It should really be `if self.size:` because bool('') is False. I think with that change it won't need another unit test.

Comment (cslzchen) [reply waterbutler/core/metadata.py]: I had thought about `if self.size` but there is one more corner case. If the size is `0`, `bool(0)` is `False` instead of true ... I guess this is the reason why we use `if self.size is not None:` in the first place.

**Comment by TomBaxter** [comment]
Merged this PR into [PR312](https://github.com/CenterForOpenScience/waterbutler/pull/312) and updated with minor logic change.

File: waterbutler/core/metadata.py (hunk 3 above)
Comment (TomBaxter) [reply waterbutler/core/metadata.py]: Completed in https://github.com/CenterForOpenScience/waterbutler/pull/312

**Comment by cslzchen** [comment]
PR Closed. As mentioned, work continues in https://github.com/CenterForOpenScience/waterbutler/pull/312.
